xrpl/core/binarycodec/types/account_id.py

```diff
@@ -41,7 +42,7 @@ def __init__(self: AccountID, buffer: Optional[bytes] = None) -> None:
             super().__init__(bytes(self.LENGTH))
 
     @classmethod
-    def from_value(cls: Type[AccountID], value: str) -> AccountID:
+    def from_value(cls: Type[Self], value: str) -> Self:
```

Comment:
what does `Type[Self]` do? 

I tried to use `cls: Self`, but MyPy complains:
```
The erased type of self "xrpl.core.binarycodec.types.account_id.AccountID" is not a supertype of its class "Type[xrpl.core.binarycodec.types.account_id.AccountID]"Mypy[misc](https://mypy.readthedocs.io/en/latest/_refs.html#code-misc)
```

I'm trying to adapt to the last example in this sub-section: https://typing.readthedocs.io/en/latest/spec/generics.html#use-in-classmethod-signatures

Reply:
Classmethods are different than normal methods. Class methods are done as `Class.method()` as opposed to `instance.method()`. Since the first parameter is the type that the method is applied on, `instance instanceof Self` and `Class instanceof Type[Self]`.